app/components/DataInput/index.js

```diff
@@ -92,80 +106,108 @@ class DataInput extends AppComponent {
     this.setState({ rawData: evt.target.value });
   }
 
-  render() {
-    let dataStatus = 'Waiting for data input';
-    let dataStatusClass = 'initial';
+  _getDataMessage() {
+    return this.props.dataStatus.message || 'Waiting for data input';
+  }
 
-    if (this.props.dataStatus) {
-      if (this.props.dataStatus.message) {
-        dataStatus = this.props.dataStatus.message;
-      }
+  _getDataClass() {
+    return this.props.dataStatus.status || 'initial';
+  }
 
-      if (this.props.dataStatus.status) {
-        dataStatusClass = this.props.dataStatus.status;
-      }
-    }
+  render() {
+    const { rawData } = this.state;
+    const {
+      canLoadSheet,
+      dateFormat,
+      firstCol,
+      metadata,
+      isNextStepAvailable,
+    } = this.props;
 
     return (
       <div className={this.parentStyles.appComponent}>
         <Heading level={2}>{appSteps[0]}</Heading>
-        <ListBlock list={this.inputRules} />
-        <ChartTitle metadata={this.props.metadata} />
-        <div>
-          <Label>Chart data</Label>
-          <textarea
-            id="DataInput"
-            className={styles.textarea}
-            value={this.state.rawData}
-            onChange={this._handleInputChange}
-            onBlur={this._handleInputBlur}
-            ref="dataInput"
-          />
-          <span className={styles[dataStatusClass]}>
-            <Text small>{dataStatus}</Text>
-          </span>
-        </div>
-
-        <div className={styles.actionsContainer}>
 
-          <div className={styles.submitContainer}>
-            <NextPrevButton
-              text="Next"
-              currentStep={0}
-              dir="next"
-              shouldEnable={this.props.isNextStepAvailable}
-              callback={this._nextCallback}
-            />
+        <div className={editorStyles.builderContainer}>
+          <div className={editorStyles.subcomponentContainer}>
+            {!!rawData && <DateFormatter
+              dateFormat={dateFormat}
+              dates={firstCol}
+            />}
+
+            {!rawData && <div>
```

Comment:
This render is HUGE,

I would probably split up each of this logically checks into separate render functions so in render you just call `{ this.renderX() }`